RobotCode/main.py

```diff
@@ -0,0 +1,247 @@
+                print("state:", state, state == State.FINDBALL)
+                #if (frame_cnt > 1000):
+                #    break
+            
+            if debug:
+                debug_frame = processedData.debug_frame
+                cv2.imshow('debug', debug_frame)
+                k = cv2.waitKey(1) & 0xff
+                if k == ord('q'):
+                    break
+            
+
+            ### Ref command read
+            if ref_enable == True:
+                try:
+                    #msg = { "signal": "start" , "targets": ["Jackbot","Jackbot"] , "baskets": ["blue","magneta"] }
+                    msg = ref.get_cmd()
+                    if msg != last_msg and msg != None:
+                            
+                        print("Message:",msg)
+                        last_msg = msg
+                        command_index = msg["targets"].index(name)
+                        if command_index > -1:
+                            command=msg["signal"]
+
+                            if command=="start":
+                                targetBasket = Color.Color.MAGENTA if msg["baskets"][command_index] == "magenta" else Color.Color.BLUE
+                                state=State.FINDBALL
+                            
+                            else:
+                                print("Command: STOP")
+                                state = State.STOPPED
+                except:    
+                    print("NO REF")
+            
+            ### Select basket
+            if targetBasket == Color.Color.BLUE:
+                ibasket = processedData.basket_b
+            else:
+                ibasket = processedData.basket_m
+            
+
+            ### Shorten stuff / Print ball location
+            try:
+                ballX=processedData.balls[-1].x
+                ballY=processedData.balls[-1].y
+                print("Ball   X:",ballX," Y:",ballY)
+            except:
+                print("Cant see a ball")
+            
+            
+            ##### STATEMACHINE #####
+
+            if state == State.STOPPED:
+                print("stopped")
+                time.sleep(0.5) 
+                continue
+
+
+            ### SPINN UNTIL BOT SEES BALL ###
+            elif state == State.FINDBALL:
+                ### robot tries to find ball
+                try:
+                    if len(processedData.balls)>0:
+                        if ballX >= middle_x - find_ball_tolerance and ballX <= middle_x + find_ball_tolerance:
+                            ### robot found ball
+                            state = State.GETCLOSE
+                            print("findball->getclose")
+                    robot.findaball(processedData)
+
+                except:
+                    print("findball error")
+
+
+            ### GET CLOSE TO THE BALL ###
+            elif state == State.GETCLOSE:
+                try:
+                    if ballY > 250:
```

Comment:
Use a ratio of screen resolution to be resolution agnostic

Reply:
Could be replaced with camera.rgb_height * 0.52, so the ball needs to be in the lower 48% of the sceen